codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -316,10 +332,10 @@ public Http2Stream setPriority(int parentStreamId, short weight, boolean exclusi
             // Already have a priority. Re-prioritize the stream.
             weight(weight);
 
-            if (newParent != parent() || exclusive) {
+            if (newParent != parent() || (exclusive && newParent.numChildren() != 1)) {
```

Comment:
@nmittler - I like it from a clarity point of view.  However I think it will be the same result in this case and it sounds like we would have to re-check some of the stuff we already know in this block (either parents are equal, or check the parent's child list, and size == 1).  If we were to add this new method do you have a preference for adding to `Http2Stream` or just private to `DefaultStream`?  The question is will this method have utility beyond this block.